samples-tests/src/test/kotlin/org/gradle/kotlin/dsl/samples/samples.kt

```diff
@@ -11,8 +12,27 @@ val samplesRootDir = File(rootProjectDir, "samples")
 
 internal
 fun copySampleProject(from: File, to: File) {
-    from.copyRecursively(to)
-    listOf(".gradle", "build").map { File(to, it) }.filter { it.exists() }.forEach {
-        it.deleteRecursively()
+    withMergedGradleProperties(to.resolve("gradle.properties")) {
+        from.copyRecursively(to)
+        listOf(".gradle", "build").map { File(to, it) }.filter { it.exists() }.forEach {
+            it.deleteRecursively()
+        }
     }
 }
+
+
+private
+fun withMergedGradleProperties(gradlePropertiesFile: File, action: () -> Unit) =
+    loadPropertiesFrom(gradlePropertiesFile).also { baseProperties ->
+        gradlePropertiesFile.delete()
```

Comment:
❓ The `gradlePropertiesFile.delete()` is confusing here. Maybe inlining the whole method would be clearer?